lib/src/lib/components/form-container/form-container.component.ts

```diff
@@ -71,24 +71,48 @@ export class Lab900Form<T> implements OnChanges {
       );
     }
     if (!changes?.data?.isFirstChange() && this.data) {
-      setTimeout(() =>
-        this.patchValues(this.data, changes?.data?.previousValue)
-      );
+      setTimeout(() => this.patchValues(this.data));
     }
   }
 
-  public patchValues(data: T, prevData?: T): void {
+  public patchValues(data: T): void {
     Object.keys(data).forEach((key: string) => {
       const control = this.form.controls[key];
-
-      if (control && isDifferent(data[key], prevData?.[key])) {
+      if (control) {
         if (control instanceof UntypedFormArray) {
           const fieldSchema = this.schema.fields.find(
             (field: Lab900FormField) => field.attribute === key
           );
           if (fieldSchema?.editType === EditType.Repeater) {
-            this.fb.createFormArray(data, fieldSchema, control);
+            const nbOfControlRows = control.controls?.length ?? 0;
+            const nbOfDataRows = data[key]?.length ?? 0;
+            if (nbOfControlRows < nbOfDataRows) {
+              for (let i = nbOfControlRows; i < nbOfDataRows; i++) {
+                control.push(
+                  this.fb.createFormGroup(
+                    fieldSchema?.nestedFields,
+                    null,
+                    data[key][i]
+                  )
+                );
+              }
+            } else if (nbOfControlRows > nbOfDataRows) {
```

Comment:
What if the lengths are equal but the data inside the row has changed?

Reply:
Then the patchdata still goes off normally, its outside of the 2 if checks